yb-voyager/src/anon/sql_anonymizer.go

```diff
@@ -396,111 +480,794 @@ func (a *SqlAnonymizer) miscellaneousNodesProcessor(msg protoreflect.Message) (e
+		if err != nil {
+			return fmt.Errorf("anon schema create: %w", err)
+		}
+
+	// ─── SCHEMA: ALTER(OWNER) ────────────────────────────────────────────
+	// SQL: ALTER SCHEMA sales_new OWNER TO sales_owner;
+	case queryparser.PG_QUERY_ALTER_OWNER_STMT_NODE:
+		ao, ok := queryparser.ProtoAsAlterOwnerStmtNode(msg)
+		if !ok {
+			return fmt.Errorf("expected AlterOwnerStmt, got %T", msg.Interface())
+		}
+
+		if ao.ObjectType != pg_query.ObjectType_OBJECT_SCHEMA {
+			return nil // not a schema DDL, skip
+		}
+
+		if ao.Object != nil && ao.Object.GetString_() != nil {
+			// anonymize the schema name
+			ao.Object.GetString_().Sval, err = a.registry.GetHash(SCHEMA_KIND_PREFIX, ao.Object.GetString_().Sval)
+			if err != nil {
+				return fmt.Errorf("anon schema alter owner: %w", err)
+			}
+		}
+
+	// ─── SCHEMA: GRANT ────────────────────────────────────────────
+	// SQL: GRANT USAGE ON SCHEMA sales TO sales_user;
+	case queryparser.PG_QUERY_GRANT_STMT_NODE:
+		gs, ok := queryparser.ProtoAsGrantStmtNode(msg)
+		if !ok {
+			return fmt.Errorf("expected GrantStmt, got %T", msg.Interface())
+		}
+
+		if gs.Objtype != pg_query.ObjectType_OBJECT_SCHEMA {
+			return nil // not a schema DDL, skip
+		}
+
+		// Anonymize each schema name in the GRANT list
+		for _, obj := range gs.Objects {
+			if str := obj.GetString_(); str != nil && str.Sval != "" {
+				str.Sval, err = a.registry.GetHash(SCHEMA_KIND_PREFIX, str.Sval)
+				if err != nil {
+					return fmt.Errorf("anon schema grant: %w", err)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (a *SqlAnonymizer) handleCollationObjectNodes(msg protoreflect.Message) (err error) {
+	switch queryparser.GetMsgFullName(msg) {
+	// ─── COLLATION: CREATE ────────────────────────────────────────────
 	/*
-		ColumnRef node is for column names in SELECT, WHERE, etc.
-		SQL:		SELECT column1 FROM sales.orders WHERE column2 = 'value';
+		SQL:  		CREATE COLLATION sales.nocase (provider = icu, locale = 'und');
+		ParseTree: 	stmt:{define_stmt:{kind:OBJECT_COLLATION defnames:{string:{sval:"sales"}} defnames:{string:{sval:"nocase"}} definition:{def_elem:{defname:"provider"
+					arg:{type_name:{names:{string:{sval:"icu"}} typemod:-1 location:43}} defaction:DEFELEM_UNSPEC location:32}} definition:{def_elem:{defname:"locale" arg:{string:{sval:"und"}} ...}}}
+	*/
+	case queryparser.PG_QUERY_DEFINE_STMT_NODE:
+		ds, ok := queryparser.ProtoAsDefineStmtNode(msg)
+		if !ok {
+			return fmt.Errorf("expected DefineStmt, got %T", msg.Interface())
+		}
+
+		if ds.Kind != pg_query.ObjectType_OBJECT_COLLATION {
+			return nil // not a collation DDL, skip
+		}
+
+		// Anonymize the qualified collation name
+		// defnames may be:
+		//   [name]
+		//   [schema, name]
+		//   [dbname, schema, name]
+		// parser won't fail if defnames has more than 3 elements, but thats not a valid SQL
+		for i, nameNode := range ds.Defnames {
+			if str := nameNode.GetString_(); str != nil && str.Sval != "" {
```

Comment:
I believe there is a helper function for this, right? Can we use that here to anonymize these defNames?